src/core/qgsgdalutils.h

```diff
@@ -63,11 +64,32 @@ class CORE_EXPORT QgsGdalUtils
     /**
      * Resamples a single band raster to the destination dataset with different resolution (and possibly with different CRS).
      * Ideally the source dataset should cover the whole area or the destination dataset.
+     *
+     * In case of different CRS, the parameter \a pszCoordinateOperation is the Proj coordinate operation string, that
+     * can be obtained with QgsCoordinateTransformContext::calculateCoordinateOperation()
+     *
      * \returns TRUE on success
      * \since QGIS 3.8
      */
     static bool resampleSingleBandRaster( GDALDatasetH hSrcDS, GDALDatasetH hDstDS, GDALResampleAlg resampleAlg, const char *pszCoordinateOperation );
 
+    /**
+     * Resamples a single band raster to the destination dataset with different resolution and different CRS.
+     * Ideally the source dataset should cover the whole area or the destination dataset.
+     *
+     * \note If possible, it is preferable to use the overload method with parameter \a pszCoordinateOperation.
+     *       But if it is not possible or it fails to obtain the Proj coordinate operation string,
+     *       this function is an alternative.
+     *
+     * \returns TRUE on success
+     * \since QGIS 3.30
+     */
+    static bool resampleSingleBandRaster( GDALDatasetH hSrcDS,
+                                          GDALDatasetH hDstDS,
+                                          GDALResampleAlg resampleAlg,
+                                          const QgsCoordinateReferenceSystem &sourceCrs,
+                                          const QgsCoordinateReferenceSystem &destinationCrs );
```

Comment:
This should also require a QgsCoordinateTransformContext argument, and ensure that IT is used to calculate the coordinate operation instead of always using the default proj operation.

Reply:
Can we make this private then? I'm concerned that someone incorrectly use this one without reading the full docs just because the signature makes it easier to call...

Reply:
By making this private, do you mean to take it out from the scope of `QgsGdalUtils`?